### PR TITLE
Upgrade to Python 3.7

### LINKED ISSUE #293: Update Python 3.6 to 3.7.12-compatible

### PR DESCRIPTION
**Pull request by TShapinsky**
Work Done:

- upgrade to ubuntu 20.04
- remove unneeded dependencies
- more deletion of files at the end of RUN statements to reduce image size
- upgrade to energyplus 9.6.0
- openstudio is now linked to the main E+ install instead of its packaged one to decrease image size
- downgrade 20.04 to python 3.7
- Build time of worker reduced by several minutes
- Fixes #293 #292 

Work To Do:

- [ ] upgrade to E+ 22.1 (need openstudio release and confirmation of spawn compatibility)
- [ ] get rid of python 3.7 (needs E+ to upgrade to 22.1)
- [ ] Merge alfalfa-client update when ready

### CODE REVIEW

**Review by nllong — APPROVED** [review]
changes look great. obviously need to figure out the test failures. happy to help if needed.

not sure what the maximum excursion error is all about ¯\_(ツ)_/¯

**Comment by TShapinsky** [comment]
> changes look great. obviously need to figure out the test failures. happy to help if needed.
> 
> not sure what the maximum excursion error is all about ¯_(ツ)_/¯

It appears to be poetry dependency hell... I'm able to recreate if I delete my lock file and try to regenerate it...

**Comment by anyaelena** [comment]
@nllong please review this PR - we want to merge.

File: alfalfa_worker/Dockerfile [hunk 1]
```diff
@@ -109,25 +94,41 @@ ENV OPENBLAS_HOME $ROOT_DIR
 
 ENV SUPERLU_HOME $ROOT_DIR/SuperLU_MT_3.1
 
+# Until we can support E+ 22.1.0 we need to install python 3.7 (thus deadsnakes/ppa)
+RUN add-apt-repository ppa:deadsnakes/ppa \
```
Comment (nllong) [inline alfalfa_worker/Dockerfile:98]: `deadsnakes`, haha, doesn't sound auspicious

**Review by nllong — APPROVED** [review]
looks good! Tests passed!

**Comment by nllong** [comment]
> @nllong please review this PR - we want to merge.

looks great! Still marked with WIP and Do Not Merge--feel free to merge if good.

**Review by nllong — APPROVED** [review]

**Pull request merged**